### PR TITLE
[WIP] ROC-1777 - Service Address

### PR DESCRIPTION
**Pull request by robertparkinson**

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/cmc/claimstore/models/ServiceAddress.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import javax.validation.constraints.Size;
+
+public class ServiceAddress {
+    @NotBlank(message = "Service Address defendantsAddress should not be empty")
+    @Size(max = 3, message = "Service Address defendantsAddress should not be longer than {max} characters")
+    private final String defendantsAddress;
+
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    @Size(max = 100, message = "Address Line1 should not be longer than {max} characters")
+    private final String line1;
```
Comment (timja-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/models/ServiceAddress.java]: Use an address class here, do not duplicate the code.
@JsonUnwrapped can be used to flatten the object so you don't get
```
serviceAddress:{
   address: {
  }
}
```

File: src/main/java/uk/gov/hmcts/cmc/claimstore/models/ServiceAddress.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            && Objects.equals(postcode, address.postcode);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(defendantsAddress, line1, line2, city, postcode);
+    }
+
+    @Override
+    public String toString() {
```
Comment (timja-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/models/ServiceAddress.java]: ReflectionToStringBuilder

File: src/main/java/uk/gov/hmcts/cmc/claimstore/models/ServiceAddress.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+import com.fasterxml.jackson.annotation.JsonInclude;
+import org.hibernate.validator.constraints.NotBlank;
+
+import java.util.Objects;
+import javax.validation.constraints.Size;
+
+public class ServiceAddress {
+    @NotBlank(message = "Service Address defendantsAddress should not be empty")
+    @Size(max = 3, message = "Service Address defendantsAddress should not be longer than {max} characters")
+    private final String defendantsAddress;
```
Comment (timja-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/models/ServiceAddress.java]: What is this field? It looks like it holds yes / no.

Comment (robertparkinson) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/models/ServiceAddress.java]: it is supposed to be a yes/no but I didn't see a class in claim store to use but I was rushed getting it in before leaving yesterday evening

Comment (timja-kainos) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/models/ServiceAddress.java]: boolean

**Pull request closed without merging**